docs/howto_contribute.md

```diff
@@ -0,0 +1,396 @@
+### Before We Get Started...
+
+This document assumes that you have some basic familiarity with the following
+topics:
+
+  - general `git` usage
+  - submitting and reviewing pull requests (PRs) on GitHub
+  - general Ansible usage (tasks, roles, playbooks)
+  - have previously read though [CONTRIBUTING.md](https://github.com/projectatomic/atomic-host-tests/blob/master/CONTRIBUTING.md)
+
+The intention is for the repo to not have any exotic requirements and be
+relatively easy to use and to contribute to.  If you feel this is not the
+case, feel free to open an [issue](https://github.com/projectatomic/atomic-host-tests/issues/new) to discuss how it can be improved.
+
+### A Brief Introduction to the Repo
+
+At the top-level of the repository, we have five directories.  Of these five
+directories, the `roles` and `tests` directories are where the majority of
+the development happens.
+
+The `callback_plugins` directory is where we have a single plugin that
+provides improved output formatting from the Ansible playbooks.  It also
+provides some functionality that allows the tests to collect the journal
+from a system after a failure.
+
+The `common` directory contains pieces of Ansible playbooks/roles that did
+not fit in with the other roles and tests that exist, but are still used
+in parts of the repository.
+
+The `docs` directory contains documentation important to using the repo
+and developing for the repo.
+
+The `roles` directory is the traditional `roles` directory that is often
+used in Ansible playbooks.  We have placed the directory at the top-level
+of the repo, so that we can re-use the roles across multiple tests.
+
+The `tests` directory contains playbooks which execute tests using a mix
+of roles and tasks.
+
+### Setting up your Environment/Tools
+
+As stated in the opening of this doc, the repo does not have any exotic
+requirements and should be easy to use and develop for.
+
+You are free to use any code editor that you are most familiar with, as long
+as it does not cause any breakage in the roles/tests or distrupts the overall
+style of the repo.
+
+That being said, `vim` users would probably be interested in the the
+[Ansible plugin](https://github.com/pearofducks/ansible-vim) which enables syntax highlighting.
+This `vim` plugin is also why many of the `.yml` files in the repo have the
+following comment at the head of file:
+
+`# vim: set ft=ansible:`
+
+### Writing Your First Role
+
+When writing a role for to be used in a test, you should attempt to make it
+fairly generic, so that it can be re-used by other users.  To show this, let's
+look at an existing role which is **NOT** generic and could be modified.
```

Comment:
Yeah, I think I can highlight what we are looking for with the just the 'fixed example' and not bothering to show the 'wrong example'.